opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -231,9 +231,8 @@ public SpanBuilder ignoreActiveSpan() {
 
         @Override
         public SpanBuilder addReference(String referenceType, SpanContext referencedContext) {
-            if (firstParent == null && (
-                    referenceType.equals(References.CHILD_OF) || referenceType.equals(References.FOLLOWS_FROM))) {
-                this.firstParent = (MockSpan.MockContext)referencedContext;
+            if (referenceType.equals(References.CHILD_OF) || referenceType.equals(References.FOLLOWS_FROM)) {
```

Comment:
Why this check? It can support arbitrary references.

Reply:
Indeed, I just kept the current behaviour. I will change this.